R/InstanceAPI.R

```diff
@@ -96,15 +91,22 @@ InstanceAPI <- R6::R6Class( # nolint object_name_linter
         body = body
       )
 
+      private$process_request(request, "get record")
+    }
+  ),
+  private = list(
+    .instance_settings = NULL,
+    process_request = function(request, request_type) {
       content <- httr::content(request)
       if (httr::http_error(request)) {
-        cli_abort(content$detail)
+        if (is.list(content) && "detail" %in% names(content)) {
+          cli_abort(content$detail)
+        } else {
+          cli_abort(paste0("Failed to ", request_type, " from instance. Response output: ", content))
+        }
```

Comment:
I'm not sure how this will work with the API package but we can sort that out later

Reply:
agreed